misk-crypto/src/main/kotlin/misk/crypto/pgp/PublicKeyProvider.kt

```diff
@@ -0,0 +1,19 @@
+package misk.crypto.pgp
+
+import com.google.inject.Provider
+import misk.crypto.Key
+import misk.crypto.KeyReader
+import org.bouncycastle.openpgp.PGPPublicKey
+import org.bouncycastle.openpgp.PGPPublicKeyRingCollection
+import org.bouncycastle.openpgp.PGPUtil
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator
+import java.io.BufferedInputStream
+
+internal class PgpPublicKeyProvider(val key: Key) : Provider<PGPPublicKey>, KeyReader() {
+  override fun get(): PGPPublicKey {
+    val keyIn = BufferedInputStream(key.encrypted_key.value.byteInputStream())
+    val pgpPub =
+        PGPPublicKeyRingCollection(PGPUtil.getDecoderStream(keyIn), JcaKeyFingerprintCalculator())
+    return pgpPub.keyRings.next().publicKey
```

Comment:
this is creating a new public key?

Reply:
nah, it just reads weird. I believe the idea is that you might have many public keys in your input stream, but in our case we'll really only ever have one so you can just call `keyRings.next().publicKey`